src/baseline.rs

```diff
@@ -182,11 +176,39 @@ fn save_placeholder_rustdoc_manifest(
     Ok(placeholder_manifest_path)
 }
 
+#[allow(dead_code)]
+pub(crate) enum CrateType<'a> {
+    Current,
+    Baseline {
+        /// When the baseline is being generated from registry
+        /// and no specific version was chosen, we want to select a version
+        /// that is the same or older than the version of the current crate.
+        highest_allowed_version: Option<&'a semver::Version>,
+    },
+}
+
+pub(crate) struct CrateDataForRustdoc<'a> {
+    pub(crate) crate_type: CrateType<'a>,
+    pub(crate) name: &'a str,
+    // TODO: pass an enum describing which features to enable
+}
```

Comment:
I'm not sure whether I like all the names here -- do you have some better ideas @obi1kenobi ?

Reply:
They're ... okay. I don't love them either but none of my ideas are substantially better, and we need this feature much more than we need fancy names here. These will do for now.